carapace-server/src/main/java/org/carapaceproxy/server/RequestsLogger.java

```diff
@@ -126,7 +130,61 @@ private void closeAccessLogFile() throws IOException {
             os = null;
         }
     }
-
+    
+    private void rotateAccessLogFile() throws IOException {
+        String accesslogPath =  this.currentConfiguration.getAccessLogPath();
+        long maxSize = this.currentConfiguration.getAccessLogMaxSize();
+        
+        File file = new File(accesslogPath);
+        
+        if(file.exists()){            
+            long currentSize = file.length();
+            
+            if( currentSize >= maxSize){
+                LOG.info("Maximum access log size reached.");
+                File newAccessLogPath = new File(accesslogPath + "-" + System.currentTimeMillis());
+                file.renameTo(newAccessLogPath);
```

Comment:
you should use the java.io.Path API (new Java IO API, introduced in Java 7 ! )
there are utilities in Files that support "rename" with ATOMIC_MOVE flag